lib/jxl/decode.cc

```diff
@@ -2646,19 +2667,49 @@ JxlDecoderStatus JxlDecoderSetImageOutCallback(JxlDecoder* dec,
                                                const JxlPixelFormat* format,
                                                JxlImageOutCallback callback,
                                                void* opaque) {
+  dec->simple_image_out_callback.callback = callback;
+  dec->simple_image_out_callback.opaque = opaque;
+  const auto init_callback =
+      +[](void* init_opaque, size_t num_threads, size_t num_pixels_per_thread) {
```

Comment:
Why are we forcing a conversion to fn pointer?

Reply:
Not sure, I have gotten into the habit of doing that when it is the end-goal, perhaps to get a compilation error as soon as possible in case of accidental capture. For what it’s worth, there are a few other places where we do this.